File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:

        count = {}

        for n in nums:
            if n not in count:
                count[n] = 1
            else:
                count[n] += 1
        

        max_val = -1
        ans = 0

        for key,val in count.items():
            if val > max_val:
                max_val = val
                ans = key
        return ans
        

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {

        Map<Integer, Integer> mapp = new HashMap<>();
        int n = nums.length;

        for(int num : nums){
            if(mapp.containsKey(num)){
                mapp.put(num, mapp.get(num) + 1);
            } else {
                map.put(num, 1);
            }
        }

        for(Map.Entry<Integer, Integer> count : mapp.entrySet()){
            Integer key = count.getKey();
            Integer val = count.getValue();

            if(val > n / 2){
                return key;
            }
        }
        
    }
}